plugin/network_processor.go

```diff
@@ -0,0 +1,258 @@
+package plugin
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/Azure/azure-sdk-for-go/profiles/latest/compute/mgmt/compute"
+	"github.com/Azure/azure-sdk-for-go/profiles/latest/network/mgmt/network"
+)
+
+type networkInfo struct {
+	IpAddresses []string
+}
+
+// Common interface for IP configuration processing
+type networkProcessor interface {
+	processIPConfigurations(ctx context.Context, ipConfigs []network.InterfaceIPConfiguration, ifName string) (*networkInfo, error)
+	getPublicIPAddress(ctx context.Context, publicIPRef *network.PublicIPAddress, ipConf *network.InterfaceIPConfiguration, ifName string) (string, error)
+	getNetworkInterface(ctx context.Context, interfaceID string) (*network.Interface, string, error)
+}
+
+// Standard VM network processor
+type vmNetworkProcessor struct {
+	clients       *azureClients
+	resourceGroup string
+}
+
+// VMSS network processor
+type vmssNetworkProcessor struct {
+	clients       *azureClients
+	resourceGroup string
+	vmssName      string
+	instanceID    string
+}
+
+func processNetworkProfile(
+	ctx context.Context,
+	interfaces []compute.NetworkInterfaceReference,
+	processor networkProcessor,
+	targetInfo *networkInfo) error {
+
+	for _, ifaceRef := range interfaces {
+		if ifaceRef.ID == nil {
+			return fmt.Errorf("nil ID for network interface")
+		}
+
+		iface, ifName, err := processor.getNetworkInterface(ctx, *ifaceRef.ID)
+		if err != nil {
+			return err
+		}
+
+		if iface.InterfacePropertiesFormat == nil || iface.InterfacePropertiesFormat.IPConfigurations == nil {
+			continue
+		}
+
+		networkInfo, err := processor.processIPConfigurations(ctx, *iface.InterfacePropertiesFormat.IPConfigurations, ifName)
+		if err != nil {
+			return err
+		}
+		targetInfo.IpAddresses = append(targetInfo.IpAddresses, networkInfo.IpAddresses...)
+	}
+	return nil
+}
+
+func (p *vmNetworkProcessor) getNetworkInterface(ctx context.Context, interfaceID string) (*network.Interface, string, error) {
+	ifResGroup, ifName, err := splitId(interfaceID, constMsNetworkService, constNetworkInterfacesResource)
+	if err != nil {
+		return nil, "", fmt.Errorf("error splitting network interface id %q: %w", interfaceID, err)
+	}
+
+	iface, err := p.clients.ifClient.Get(ctx, ifResGroup, ifName, "")
+	if err != nil {
+		return nil, "", fmt.Errorf("error fetching network interface: %w", err)
+	}
+
+	return &iface, ifName, nil
+}
+
+func (p *vmNetworkProcessor) processIPConfigurations(
+	ctx context.Context,
+	ipConfigs []network.InterfaceIPConfiguration,
+	ifName string) (*networkInfo, error) {
```

Comment:
Ditto for this and others. My rule is, either all input parameters on one line, or each parameter gets its own line. Same for result parameters.
```suggestion
	ifName string,
) (*networkInfo, error) {
```

Reply:
Ok, lovely stuff - I've been back through the files I've added / changed and reformatted the input and output parameters according to your suggestions 👍